### ISSUE TITLE
NAPALM proxy minion should generate ipv4/6 grain

### ISSUE THREAD

**Issue by ngrundler** [saltstack/salt#47996]
### Description of Issue/Question
The NAPALM proxy minion currently does not set the ipv4/6 grain, which is required for using `ipcidr` targetting.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
Given a standard proxy key:
```
proxy:
  proxytype: napalm
  driver: junos
  host: 10.0.0.1
  username: juniper
  passwd: juniper
```

### Steps to Reproduce Issue
$ salt -S '10.0.0.1' test.ping

No minions matched the target. No command was sent, no jid was assigned.                                              
ERROR: No return received 

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

Salt Version:                                                                                                         
           Salt: 2018.3.0                                                                                             
                                                                                                                      
Dependency Versions:                                                                                                  
           cffi: 1.11.4                                                                                               
       cherrypy: unknown                                                                                              
       dateutil: 1.5                                                                                                  
      docker-py: 1.10.6                                                                                               
          gitdb: 0.6.4
      gitpython: 1.0.1                                                                                                
          ioflo: Not Installed                                                                                        
         Jinja2: 2.7.2                                                                                                
        libgit2: Not Installed                                                                                        
        libnacl: Not Installed                                                                                        
       M2Crypto: 0.28.2                                                                                               
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: 3.5.1
         pygit2: Not Installed
         Python: 2.7.5 (default, May  3 2017, 07:55:04)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: redhat 7.4 Maipo
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.21.1.el7.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 7.4 Maipo

**Comment by gtmanfred** [comment]
Assigned to @mirceaulinic as discussed in the networktocode slack.

**Comment by mirceaulinic** [comment]
In 2017.7.0 I added this grain: https://docs.saltstack.com/en/latest/ref/grains/all/salt.grains.napalm.html#salt.grains.napalm.host_dns, so this information exists, but:

- Not under the right grain.
- Not enabled by default (as Mike and Seth suggested in https://github.com/saltstack/salt/pull/41546#issuecomment-305846883)
- What should be the target branch for this fix? `2017.7`?

Thanks.

**Comment by gtmanfred** [comment]
yes, fixing in the 2017.7 branch seems like the right place.
